compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -285,6 +286,10 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                                 bug!("Only valid to do a DynStar cast into a DynStar type")
                             };
                         let vtable = get_vtable(bx.cx(), source.ty(self.mir, bx.tcx()), trait_ref);
+                        let data = match operand.layout.pointee_info_at(bx.cx(), Size::ZERO) {
```

Comment:
It seems like there ought to be a better way to check if an operand is a pointer.

Also, this will likely be unnecessary once we have the conversion traits.

Reply:
Yeah pointee_info is definitely the wrong thing to check here. That type is for optimization hints only.

You have the full type layout here, you can just check `layout.abi()`.